FnCAS/fncas/mathutil.h

```diff
@@ -135,8 +153,9 @@ inline std::vector<double> Backtracking(F&& eval_function,
       break;
     }
   }
+  OptimizerLogger().Log(std::string("Backtracking: Final value = ") + current::ToString(test_f_value));
 
-  return test_point;
+  return std::make_pair(test_f_value, test_point);
```

Comment:
A couple of thoughts:
`OptimizationResult` has the same structure. We can unify them under some `PointAndValue` structure.
We also use `std::vector<double>` explicitly to define a point in R<sup>n</sup>. As we go further with more sophisticated linear algebra methods, a lack of proper mathematical vector/matrix notation could become troublesome. I don't suggest to spend time on linear algebra engine now, but maybe it's better to add some `Point` class (aliased to `std::vector<double>`) abstraction at this stage, so we can easily upgrade math engine later. What fo you think?